Improve instructions and fix tests for PyhagoreanTriplet exercise.

LarsWestergren opened the pull request:
Fixes https://github.com/exercism/scala/issues/636

ErikSchierboom commented on exercises/practice/pythagorean-triplet/.docs/instructions.md:
````diff
@@ -16,9 +16,15 @@ a < b < c
+3² + 4² = 9 + 16 = 25 = 5².
 ```
 
+Implement the method `isPythagorean((Int, Int, Int)): Boolean` which checks the given triplet. 
+
+Then implement `pythagoreanTriplets(min: Int, max: Int): Seq[(Int, Int, Int)]` which should returns a sequence of all
+triplets whose sides are in the range `min` to `max` inclusive
+
+Finally implement `pythagoreanTripletsSum(n: Int): Seq[(Int, Int, Int)]`
````
Editing this file unfortunately is discouraged. That is due to the contents of the file being synced from a shared exercise source, in this case: https://github.com/exercism/problem-specifications/blob/main/exercises/pythagorean-triplet/description.md

If you need to add track-specific instructions, see https://exercism.org/docs/building/tracks/practice-exercises#h-file-docs-instructions-append-md

LarsWestergren replied:
Ah, I see. I was a bit surprised that descriptions weren't centralized, but since the Scala docs differed...

Reverting. I hope the easier to read "3²" will be synced to the Scala repo after my commit is merged.

ErikSchierboom replied:
You could do this yourself via configlet. First run `./bin/fetch-configlet`. Then do: `./bin/configlet sync --docs --exercise pythagorean-triplets --update --yes`. We can also do this later 🤷

LarsWestergren commented on exercises/practice/pythagorean-triplet/src/main/scala/PythagoreanTriplet.scala:
```diff
@@ -0,0 +1,5 @@
+object PythagoreanTriplet {
+  def isPythagorean(nums: (Int, Int, Int)): Boolean = ???
+  def pythagoreanTriplets(low: Int, high:Int): Seq[(Int, Int, Int)] = ???
+  def pythagoreanTripletsSum(p: Int): Seq[(Int, Int, Int)] = ???
```
I see. Clarification of expectations since it seemed confusing. But this feels like a breaking change. I will revert this and fix the tests instead. It seems all other languages I check do not test the ordering of the 3 arguments, so it is up to the implementer to decide if this is a requirement to pass or not. Both interpretations are certainly possible.

LarsWestergren commented:
Ready for a new review, sorry for all the commits.

After checking several other languages (Go, Java, F#, Ruby, Haskell) they all have tests for the method/function the problem mainly discusses - "Given an input integer N, find all Pythagorean triplets", so I added tests and an implementation for that. Having it makes it clearer what the other methods do, so instructions shouldn't need changing.

Removed the automatic sorting of inputs in .meta/Example.scala.

Should I squash all the commits into one?

ErikSchierboom commented:
@LarsWestergren CI is failing. It looks to be related to the changes in this PR. Could you look into it?

LarsWestergren commented:
> @LarsWestergren CI is failing. It looks to be related to the changes in this PR. Could you look into it?

Embarassing, missed comitting my last change.  Fixed now, and also used configlet to sync the docs.

ErikSchierboom commented:
Merged! Thanks 🎉
